File: Modules/ServerTypes/hospital/DoctorsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DocSpecialityListFormatter } from "../../hospital/Doctors/DocSpecialityListFormatter";
import { DoctorsRow } from "./DoctorsRow";

export interface DoctorsColumns {
    DoctorId: Column<DoctorsRow>;
    DoctorName: Column<DoctorsRow>;
    SpecialityList: Column<DoctorsRow>;
}

export class DoctorsColumns extends ColumnsBase<DoctorsRow> {
    static readonly columnsKey = 'hospital.Doctors';
    static readonly Fields = fieldsProxy<DoctorsColumns>();
}

[DocSpecialityListFormatter]; // referenced types

File: Modules/hospital/Doctors/DoctorsColumns.cs
using hospital;
using MovieTutorial;

namespace Serene2.hospital.Columns;

[ColumnsScript("hospital.Doctors")]
[BasedOnRow(typeof(DoctorsRow), CheckNames = true)]
public class DoctorsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignCenter]
    public int DoctorId { get; set; }
    [EditLink]
    public string DoctorName { get; set; }

    [Width(200), DocSpecialityListFormatter]
    public List<int> SpecialityList { get; set; }

}
